File: LabourAssist/app/src/main/java/com/example/labourassist/MainActivity.java
package com.example.labourassist;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.RecyclerView;
import androidx.viewpager.widget.ViewPager;
import androidx.viewpager2.widget.ViewPager2;

import android.os.Bundle;
import android.widget.Adapter;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {
    ViewPagerAdapter adapter;ViewPager2 viewPager2;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        viewPager2=findViewById(R.id.viewpager);
        List<Integer> images=new ArrayList<Integer>();
        images.add(R.drawable.imga); images.add(R.drawable.imgb); images.add(R.drawable.imgc);
      adapter = new ViewPagerAdapter(images);
        viewPager2.setAdapter(adapter);

    }
}

File: TrainMvvp/app/src/main/java/com/example/trainmvvp/MainActivity.kt
package com.example.trainmvvp

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

class MainActivity : AppCompatActivity() {
    lateinit var viewModel: NoteViewModel
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        viewModel=ViewModelProvider(this,
        ViewModelProvider.AndroidViewModelFactory.getInstance(application)).get(NoteViewModel::class.java)
        viewModel.allNotes.observe(this, Observer {

        })
       recyclerView.layoutManager=LinearLayoutManager(this)
        val adapter=NotesRvAdapter(this,this)
    }
}